dspace-server-webapp/src/test/java/org/dspace/app/rest/SitemapRestControllerIT.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.dspace.app.rest.builder.CollectionBuilder.createCollection;
+import static org.dspace.app.rest.builder.CommunityBuilder.createCommunity;
+import static org.dspace.app.rest.builder.ItemBuilder.createItem;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import javax.servlet.ServletException;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MvcResult;
+
+/**
+ * Integration test to test the /sitemaps/{name} endpoint, see {@link SitemapRestController}
+ *
+ * @author Maria Verdonck (Atmire) on 08/07/2020
+ */
+public class SitemapRestControllerIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    ConfigurationService configurationService;
+
+    private final static String SITEMAPS_ENDPOINT = "sitemaps";
+
+    private Item item1;
+    private Item item2;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        configurationService.setProperty("sitemap.path", SITEMAPS_ENDPOINT);
+
+        context.turnOffAuthorisationSystem();
+
+        Community community = createCommunity(context).build();
+        Collection collection = createCollection(context, community).build();
+        this.item1 = createItem(context, collection)
+            .withTitle("Test 1")
+            .withIssueDate("2010-10-17")
+            .build();
+        this.item2 = createItem(context, collection)
+            .withTitle("Test 2")
+            .withIssueDate("2015-8-3")
+            .build();
+
+        runDSpaceScript("generate-sitemaps");
+
+        context.restoreAuthSystemState();
+    }
+
+    @After
+    public void destroy() throws Exception {
+        // delete sitemaps generated by tests in before
+        runDSpaceScript("generate-sitemaps", "-d");
```

Comment:
can we call the static method directly here? it would be more easy to understand and efficient

Reply:
Do you have an example of the alternative method you're asking for? I didn't see any problems with using the DSpace script

Reply:
I simply mean to call directly `org.dspace.app.sitemap.GenerateSitemaps.deleteSitemaps()` as it is a static public method